### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/ArrayUtils/MergeReplaceKey.php [hunk 1]
```diff
@@ -6,15 +6,11 @@
...
 final class MergeReplaceKey implements MergeReplaceKeyInterface
 {
-    /** @var mixed */
-    protected $data;
-
     /**
      * @param mixed $data
      */
-    public function __construct($data)
+    public function __construct(protected $data)
```
Comment (Ocramius) [inline src/ArrayUtils/MergeReplaceKey.php]: `mixed` can be explicit here

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ArrayUtils/MergeReplaceKey.php (hunk 1 above)
Comment (samsonasik) [reply src/ArrayUtils/MergeReplaceKey.php]: Ok, I applied `mixed` param type on https://github.com/laminas/laminas-stdlib/pull/75/commits/d1085083254efd023f628338ccb1fe2a8ca00a26

**Comment by samsonasik** [comment]
All green 🎉

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**